packages/nest-api/src/keyword-scraping-module/keyword-record.controller.spec.ts

```diff
@@ -0,0 +1,214 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { KeywordRecordController } from './keyword-record.controller';
+import { KeywordRecordService } from './services/keyword-record.service';
+import { KeywordRecordSearchQueryDto } from './dto/keyword-record-search-query.dto';
+import { KeywordRecord } from './models/keyword-record.model';
+import { StreamableFile } from '@nestjs/common';
+import { ScrapeJobDonePayload } from '../core/types';
+import { RmqContext } from '@nestjs/microservices';
+
+jest.mock('./services/keyword-record.service');
+
+describe('AuthController', () => {
+  let controller: KeywordRecordController;
+  let keywordRecordService: KeywordRecordService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      controllers: [KeywordRecordController],
+      providers: [KeywordRecordService],
+    }).compile();
+
+    controller = module.get<KeywordRecordController>(KeywordRecordController);
+    keywordRecordService =
+      module.get<KeywordRecordService>(KeywordRecordService);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+    expect(keywordRecordService).toBeDefined();
+  });
+
+  describe('listAllKeywordRecords', () => {
+    it('should return list of keyword records with paginate', async () => {
+      const req = {
+        user: jest.fn(),
+      };
+      const dto = new KeywordRecordSearchQueryDto();
+      const mockValue = {
+        data: [],
+        paginate: { total: 0, per_page: 0, current_page: 0 },
+      };
+
+      jest
+        .spyOn(keywordRecordService, 'getListData')
+        .mockReturnValueOnce(Promise.resolve(mockValue));
+
+      const result = await controller.listAllKeywordRecords(req, dto);
+
+      expect(result).toEqual(mockValue);
+    });
+  });
+  describe('uploadKeywordRecords', () => {
```

Comment:
small thing: there should be a line break to make the code easier to read and consistent with other test files 😃
```suggestion
  });
  
  describe('uploadKeywordRecords', () => {
````